qiskit/synthesis/multi_controlled/mcx_synthesis.py

```diff
@@ -300,6 +301,308 @@ def synth_mcx_noaux_v24(num_ctrl_qubits: int) -> QuantumCircuit:
     return qc
 
 
+def linear_depth_ladder_ops(qreg: List[int]) -> Tuple[QuantumCircuit, List[int]]:
+    r"""
+    Helper function to create linear-depth ladder operations used in Khattar and Gidney's MCX synthesis.
+    In particular, this implements Step-1 and Step-2 on Fig. 3 of [1] except for the first and last
+    CCX gates.
+
+    Args:
+        qreg: List of qubit indices to apply the ladder operations on. qreg[0] is assumed to be ancilla.
+
+    Returns:
+        QuantumCircuit: Linear-depth ladder circuit.
+        int: Index of control qubit to apply the final CCX gate.
```

Comment:
Since the signature has typehints, you can omit them here.
```suggestion
        A tuple consisting of the linear-depth ladder circuit and the index of 
        control qubit to apply the final CCX gate.
```